test/resolve-config.test.js

```diff
@@ -41,102 +56,158 @@ test('Returns user config via environment variables', t => {
       assets,
       milestones,
       successComment: undefined,
+      proxy,
     }
   );
 });
 
-test('Returns user config via alternative environment variables', t => {
-  const gitlabToken = 'TOKEN';
-  const gitlabUrl = 'https://host.com';
-  const gitlabApiPathPrefix = '/api/prefix';
-  const assets = ['file.js'];
+test("Returns user config via alternative environment variables", (t) => {
+  const gitlabToken = "TOKEN";
+  const gitlabUrl = "https://host.com";
+  const gitlabApiPathPrefix = "/api/prefix";
+  const assets = ["file.js"];
+  const proxy = {};
 
   t.deepEqual(
-    resolveConfig({assets}, {env: {GL_TOKEN: gitlabToken, GL_URL: gitlabUrl, GL_PREFIX: gitlabApiPathPrefix}}),
+    resolveConfig({ assets }, { env: { GL_TOKEN: gitlabToken, GL_URL: gitlabUrl, GL_PREFIX: gitlabApiPathPrefix } }),
     {
       gitlabToken,
       gitlabUrl,
       gitlabApiUrl: urlJoin(gitlabUrl, gitlabApiPathPrefix),
       assets,
       milestones: undefined,
       successComment: undefined,
+      proxy,
     }
   );
 });
 
-test('Returns default config', t => {
-  const gitlabToken = 'TOKEN';
-  const gitlabApiPathPrefix = '/api/prefix';
-  const gitlabUrl = 'https://gitlab.com';
+test("Returns user config via alternative environment variables with http proxy", (t) => {
+  const gitlabToken = "TOKEN";
+  const gitlabUrl = "http://host.com";
+  const gitlabApiPathPrefix = "/api/prefix";
+  const assets = ["file.js"];
+  // Testing with 8080 port because HttpsProxyAgent ignores 80 port with http protocol
+  const proxyUrl = "http://proxy.test:8080";
+
+  const result = resolveConfig(
+    { assets },
+    {
+      env: {
+        GL_TOKEN: gitlabToken,
+        GL_URL: gitlabUrl,
+        GL_PREFIX: gitlabApiPathPrefix,
+        HTTP_PROXY: proxyUrl,
+      },
+    }
+  );
+
+  t.assert(result.proxy.agent.http instanceof HttpProxyAgent);
+  t.assert(result.proxy.agent.http.proxy.origin === proxyUrl);
+});
+
+test("Returns user config via alternative environment variables with https proxy", (t) => {
```

Comment:
Can we add similar test cases for situations where we expect no or a default agent?

- HTTPS GitLab URL and HTTP_PROXY set
- HTTP GitLab URL and HTTPS_PROXY set
- Neither HTTP_PROXY nor HTTPS_PROXY set

Reply:
Done.

- Neither HTTP_PROXY nor HTTPS_PROXY set
The above is covered by the test on Line 6 by not passing a proxy config or passing in an empty one. 

Both cases return an empty object. Same when gitlab and proxy urls mismatch.